### PR TITLE
fix(protractor): Shows locator used when a timeout occur.

### LINKED ISSUE #2996: More descriptive errors for element.all(locator) when waitForAngular fails.
Feature Request:
**tl;dr:** Add the locator to error messages in waitForAngular.

If the waitForAngular call times out when calling element.all(locator) we get an error message like the following:

`Timed out waiting for Protractor to synchronize with the page after 9 seconds.
Please see https://github.com/angular/protractor/blob/master/docs/faq.md. 
The following tasks were pending:
    - $timeout: function o(){var e=n&&n.$$destroyed,t=e?[]:r.queue,o=e?null:r.digest;r.queue=[],r.timeout=null,r.digest=!1,t.forEach(function(e){e()}),o&&d.$digest()}
`
Unfortunately, we don't have any information about the specific locator which initiated the waitForAngular call. Should be easy enough to pass the locator as the description here:
https://github.com/angular/protractor/blob/master/lib/element.js#L133

I would love to have an error message like this:

`Timed out waiting for Protractor to synchronize with the page after 9 seconds. 
Please see https://github.com/angular/protractor/blob/master/docs/faq.md.
Locator: by.css('.parent'). 
The following tasks were pending:
    - $timeout: function o(){var e=n&&n.$$destroyed,t=e?[]:r.queue,o=e?null:r.digest;r.queue=[],r.timeout=null,r.digest=!1,t.forEach(function(e){e()}),o&&d.$digest()}
`
- Reasons for adopting new feature
Improve error clarity and ease of debugging.
 
- Is this a breaking change? (How will this affect existing functionality)
No. Should be really easy to implement too!

### PR DESCRIPTION
**Pull request by luishvz**
Fixes #2996

### CODE REVIEW

**Comment by luishvz** [comment]
Signed it!

File: lib/protractor.js [hunk 1]
```diff
@@ -401,8 +401,8 @@ Protractor.prototype.waitForAngular = function(opt_description) {
       }
       if (timeout) {
         var errMsg = 'Timed out waiting for Protractor to synchronize with ' +
-            'the page after ' + timeout + '. Please see ' +
-            'https://github.com/angular/protractor/blob/master/docs/faq.md';
+        'the page after ' + timeout + '. Locator: ' + description + '. Please see ' +
```
Comment (sjelin) [inline lib/protractor.js]: Maybe some newlines here?  Also, the description won't always be a locator

File: lib/element.js [hunk 2]
```diff
@@ -130,7 +130,7 @@ ElementArrayFinder.prototype.all = function(locator) {
   var getWebElements = function() {
     if (self.getWebElements === null) {
       // This is the first time we are looking for an element
-      return ptor.waitForAngular().then(function() {
+      return ptor.waitForAngular(locator).then(function() {
```
Comment (sjelin) [inline lib/element.js]: `opt_description` is supposed to be a string.  Maybe instead it should be `waitForAngular('Locator: ' + locator)`?

**Comment by luishvz** [comment]
@sjelin modified the parameter sent to waitForAngular() and now the `description` is only added to error message when is a locator.

`Failed: Timed out waiting for Protractor to synchronize with the page after 11 seconds. Please see https://github.com/angular/protractor/blob/master/docs/faq.md
     - Locator: by.binding("slowAngularTimeoutPromiseStatus"). 
    The following tasks were pending:
     - $timeout: function () {
          // intentionally empty
        }
`

**Comment by cnishina** [comment]
closed with b6f1a63da77ec88a3f487e5a099df8febe3742aa

**Comment by juliemr** [comment]
For the future - changes like this are not a 'fix', since it's not fixing a bug. I would call it a 'feat' or 'chore'.

**Comment by luishvz** [comment]
got it @juliemr , thanks.

**Pull request closed without merging**